lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,360 @@
+          Padding(
+            padding: const EdgeInsetsDirectional.fromSTEB(5, 3, 4, 3),
+            child: Text(user?.fullName ?? zulipLocalizations.unknownUserName,
+              style: TextStyle(
+                fontSize: 16,
+                height: 16 / 16,
+                color: designVariables.labelMenuButton))),
+        ]));
+  }
+}
+
+class _NewDmUserList extends StatelessWidget {
+  const _NewDmUserList({
+    required this.filteredUsers,
+    required this.selectedUserIds,
+    required this.onUserTapped,
+  });
+
+  final List<User> filteredUsers;
+  final Set<int> selectedUserIds;
+  final void Function(int userId) onUserTapped;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    if (filteredUsers.isEmpty) {
+      return Center(
+        child: Padding(
+          padding: const EdgeInsets.all(24),
+          child: Text(
+            zulipLocalizations.newDmSheetNoUsersFound,
+            style: TextStyle(
+              color: designVariables.labelMenuButton,
+              fontSize: 16))));
+    }
+
+    return ListView.builder(
+      padding: EdgeInsets.all(8),
+      itemCount: filteredUsers.length,
+      itemBuilder: (context, index) {
+        final user = filteredUsers[index];
+        final isSelected = selectedUserIds.contains(user.userId);
+
+        return InkWell(
+          splashFactory: NoSplash.splashFactory,
+          onTap: () => onUserTapped(user.userId),
+          child: Container(
+            decoration: isSelected ? BoxDecoration(
+              color: designVariables.bgMenuButtonSelected,
+              borderRadius: BorderRadius.circular(10)) : null,
+            child: Padding(
+              padding: const EdgeInsets.fromLTRB(0, 6, 12, 6),
+              child:Row(
+                children: [
+                  Padding(
+                    padding: const EdgeInsets.symmetric(horizontal: 8),
+                    child: Icon(
+                      isSelected ? Icons.check_circle : Icons.circle_outlined,
+                      color: isSelected
+                        ? designVariables.radioFillSelected
+                        : designVariables.radioBorder,
+                      size: 24)),
+                  Avatar(userId: user.userId, size: 32, borderRadius: 3),
+                  Expanded(
+                    child: Padding(
+                      padding: const EdgeInsetsDirectional.fromSTEB(8, 6.5, 0, 6.5),
+                      child: Text(store.userDisplayName(user.userId),
+                        style: TextStyle(
+                          fontSize: 17,
+                          height: 19 / 17,
+                          color: designVariables.textMessage)
+                            .merge(weightVariableTextStyle(context, wght: 500))))),
```

Comment:
nit:
```suggestion
                          color: designVariables.textMessage,
                        ).merge(weightVariableTextStyle(context, wght: 500))))),
```
This way the end boundary of `TextStyle` is clear.